easybuild/easyconfigs/g/Gradio/Gradio-4.19.0-foss-2023a.eb

```diff
@@ -0,0 +1,123 @@
+easyblock = 'PythonBundle'
+
+name = 'Gradio'
+version = '4.19.0'
+
+homepage = "https://www.gradio.app"
+description = """
+Gradio is an open-source Python package that allows you to quickly build a demo or web
+application for your machine learning model, API, or any arbitrary Python function. 
+"""
+
+toolchain = {'name': 'foss', 'version': '2023a'}
+
+builddependencies = [
+    ('binutils', '2.40'),
+    ('Rust', '1.75.0'),
+    ('maturin', '1.4.0', '-Rust-1.75.0'),
+]
+
+dependencies = [
+    ('Python', '3.11.3'),
+    ('SciPy-bundle', '2023.07'),
+    ('PyYAML', '6.0'),
+    ('pydantic', '2.5.3'),
+    ('matplotlib', '3.7.2'),
+    ('Pillow', '10.0.0'),
+    ('tqdm', '4.66.1'),
+    ('hatchling', '1.18.0'),
+    ('poetry', '1.5.1'),
```

Comment:
`poetry` can be just a build dependency

in principle, same for `hatchling`, but I see that `pip check` on `hatch-requirements-txt` then fails. I suppose that in itself should be a build dependency, but for that we'd need to either create an easyconfig for it or add it to the easyconfig for `hatching`...

Reply:
`hatch-requirements-txt` is a hatchling plugin, so it belongs there imho

We definitely want to avoid dragging `hatching` in as a runtime dependency, there's a reason it was split off into a standalone easyconfig...

Reply:
@WilleBell once https://github.com/easybuilders/easybuild-easyconfigs/pull/20389 is merged, you'll be able to remove `hatch-requirements-txt` from this easyconfig and also move `hatchling` to `builddependencies`, besides `poetry`